tmail-backend/event-bus-redis/src/main/java/org/apache/james/events/CleanRedisEventBusService.java

```diff
@@ -0,0 +1,147 @@
+package org.apache.james.events;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.apache.james.util.MDCStructuredLogger;
+import org.apache.james.util.ReactorUtils;
+import org.apache.james.util.StructuredLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.lettuce.core.KeyScanArgs;
+import io.lettuce.core.KeyScanCursor;
+import io.lettuce.core.ScanCursor;
+import io.lettuce.core.api.reactive.RedisKeyReactiveCommands;
+import io.lettuce.core.api.reactive.RedisSetReactiveCommands;
+import io.lettuce.core.pubsub.api.reactive.RedisPubSubReactiveCommands;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CleanRedisEventBusService {
+    public static class Context {
+        private final AtomicLong totalBindings;
+        private final AtomicLong danglingBindings;
+        private final AtomicLong cleanedBindings;
+
+        public Context() {
+            this.totalBindings = new AtomicLong();
+            this.cleanedBindings = new AtomicLong();
+            this.danglingBindings = new AtomicLong();
+        }
+
+        void increaseScannedBindingsCount() {
+            totalBindings.incrementAndGet();
+        }
+
+        void increaseDanglingBindingsCount(long count) {
+            danglingBindings.addAndGet(count);
+        }
+
+        void increaseCleanedBindingsCount(long count) {
+            cleanedBindings.addAndGet(count);
+        }
+
+        public long getTotalBindings() {
+            return totalBindings.get();
+        }
+
+        public long getDanglingBindings() {
+            return danglingBindings.get();
+        }
+
+        public long getCleanedBindings() {
+            return cleanedBindings.get();
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CleanRedisEventBusService.class);
+
+    private final RedisSetReactiveCommands<String, String> redisSetCommands;
+    private final RedisPubSubReactiveCommands<String, String> redisPubSubCommands;
+    private final RedisKeyReactiveCommands<String, String> redisKeyCommands;
+    private final RoutingKeyConverter routingKeyConverter;
+    private Context context;
+
+    @Inject
+    @Singleton
```

Comment:
Maybe we don't need "bean" for it. 
We only use it one time when start up

Reply:
Ok. It would be better for memory as the bean is likely a long-lived object.